scripts/template_matcher.py

```diff
@@ -1,13 +1,16 @@
-import cv2
+#!/usr/bin/env python 
 
 """
 This code determines which of a set of template images matches
 an input image the best using the SIFT algorithm
 """
 
+import cv2
+import numpy as np
+
 class TemplateMatcher(object):
 
-    def __init__ (self, images, min_match_count=10, good_thresh=0.7):
+    def __init__ (self, images, min_match_count=10, good_thresh=0.5):
```

Comment:
According to Figure 11 in Lowe's SIFT Paper (https://www.cs.ubc.ca/~lowe/papers/ijcv04.pdf), good_thresh aka ratio_threshold is set to 0.75.  By setting it lower, you are theoretically throwing away more true positives, while excluding a negligible amount of false positives.  I saw from your GIF that your image after the warpedTransform (img_T) was flipping all about; I wonder if this ratio threshold has anything to do with that.